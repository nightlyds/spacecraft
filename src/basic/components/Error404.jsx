import React, { useState } from "react";
import counterpart from "counterpart";
import "../css/basic.css";
import "../css/media_basic.css";
import en from "../lang/en";
import uk from "../lang/uk";
import MenuButton from "./menu/MenuButton.jsx";
import Menu from "./menu/Menu.jsx";
import Footer from "./Footer.jsx";
import LanguageChange from "./LanguageChange.jsx";
import Translate from 'react-translate-component';

counterpart.registerTranslations("en", en);
counterpart.registerTranslations("uk", uk);

const Error404 = () => {
  let [menuOpenCheck, setMenuOpen] = useState(false);

  function menuOpen() {
    setMenuOpen(!menuOpenCheck);
  }
  return (
    <div className="error-404">
      <MenuButton menuOpen={menuOpen} icon={menuOpenCheck} />
      {menuOpenCheck && <Menu />}
      <LanguageChange />
      <div className="error-404-content">
        <h2 className="error-404-title">404</h2>
        <Translate component="span" content="error404" className="error-404-text" />
      </div>
      <Footer />
    </div>
  );
};

export default Error404;
