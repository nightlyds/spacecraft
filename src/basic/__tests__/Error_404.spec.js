import React from 'react';
import Error404 from '../components/Error404.jsx';
import { shallow } from 'enzyme';

const setUp = (props) => shallow(<Error404 {...props} />);

describe("Language Change component tests", () => {
    let component;

    beforeEach(() => {
        component = setUp();
    })
    
    it("Snapshot test", () => {
        expect(component).toMatchSnapshot();
    })
})