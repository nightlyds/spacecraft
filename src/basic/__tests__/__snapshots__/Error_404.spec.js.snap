// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Change component tests Snapshot test 1`] = `
<div
  className="error-404"
>
  <MenuButton
    icon={false}
    menuOpen={[Function]}
  />
  <LanguageChange />
  <div
    className="error-404-content"
  >
    <h2
      className="error-404-title"
    >
      404
    </h2>
    <Translate
      className="error-404-text"
      component="span"
      content="error404"
    />
  </div>
  <Footer />
</div>
`;
